server/opendp_apps/analysis/release_info_formatter.py

```diff
@@ -41,15 +53,32 @@ def get_readable_datetime(self, dt_obj: dt):
                        + tz_str
         return readable_str
 
+    def get_release_data(self, as_json=False):
+        """Return the formatted release"""
+        assert not self.has_error(), \
+            "Make sure `.has_error() if False` before calling .get_release_data()"
+
+        if as_json is True:
+            return json.dumps(self.release_dict, indent=4)
+
+        return self.release_dict
+
+
 
-    def get_release_data(self):
-        """Return the release!"""
+    def build_release_data(self):
+        """Build the release!"""
         current_dt = dt.now()
 
-        self.release_dict = \
-        {
+        ds_formatter = DataSetFormatter(self.dataset)
+        if ds_formatter.has_error():
+            self.add_err_msg(ds_formatter.get_err_msg())
+            return
+        else:
+            dataset_dict = ds_formatter.get_formatted_info()
+
+        self.release_dict = {
             "name": str(self.release_util.analysis_plan),
-            "release_url": '(not available, should be url within DP Creator)',
+            "release_url": None,    # to do...
```

Comment:
What do we need to do for the release URL? Should we do it before merging this?

Reply:
I moved it to this issue #34 and put comments in the code linking to the issue.
e.g. Probably enough for a separate ticket/endpoint. Is it a "public" detail page? File/file-like objects aren't handled through regular MEDIA_URL, etc and need some extra code